ipaserver/plugins/baseruserfas.py

```diff
@@ -7,12 +7,34 @@
 
 Common user extensions
 """
+import re
+
 from ipalib import _
-from ipalib.parameters import DateTime, Str
+from ipalib.parameters import DateTime, Str, StrEnum
 
 from ipaserver.plugins.baseuser import baseuser
 from ipaserver.plugins.internal import i18n_messages
 
+# preferred pronouns
+PRONOUNS = {
+    # https://uwm.edu/lgbtrc/support/gender-pronouns/
+    "ae/aer": _("ae/aer"),
+    "fae/faer": _("fae/faer"),
+    "e/em": _("e/em"),
+    "ey/em": _("ey/em"),
+    "he/him": _("he/him"),
+    "per/per": _("per/per"),
+    "she/her": _("she/her"),
+    "they/them": _("they/them"),
+    "ve/ver": _("ve/ver"),
+    "xe/xem": _("xe/xem"),
+    "ze/hir": _("ze/hir"),
+    "zie/hir": _("zie/hir"),
```

Comment:
I suspect translation might be problematic in the long run. 

For example, french do not have a neutral pronoun, there is no agreement on which to use among LGBT+ folks (between "iel", "ielle", "yel", "yelle"), and like in English, people are also proposing others stuff  (ul, ol, al) either informally (see https://www.academia.edu/33393890/Innovations_in_Gender-Neutral_French_Language_practices_of_nonbinary_French_speakers_on_Twitter), or formally (https://simonae.fr/sciences-culture/litterature/nommer-exister-alpheratz-troisieme-genre/ in french, a article on the thesis of a linguist on the neo pronoun "al"). So far, I see "iel" being used in LGBTQ+ circles, but that's quite limited.

So we wouldn't have a mapping between anything in English and in French, except she/he, which mean that this part of the interface might always be less than 90% translated in French, which would be confusing, and not great for people who do use "iel" or any others pronouns.

Also, there is language without grammatical gender (Japanese, Turkish, Hungarian to name some), so I am not sure exactly what would the interface be like in those languages. 

If we do assume people are speaking English for Fedora participation, that's likely going to cover 98% of the community, but maybe we do also have people who do not, and who open a account for localized participation (forum/support, etc) that do not requires English.
 
How much flexibility do we have when it come to the interface ? (like how would we add localized help ?)

Reply:
Since it's possible for the user to add custom values, maybe only give the following choices:
- she/her
- he/him
- they/them

What do you think? What about adding a 4th choice "Ask me", for those who prefer their interlocutor to ask them when they interact with them? (maybe as a preference or because they are genderfluid)

Reply:
OK, we can start simpler and expend the list later. My intention was to nudge users to grab a pre-defined value. If most users use one of the pre-defined values, then it's easier to locate deviations and look for CoC violators. Or maybe I'm just overengineering stuff here :)